### DISCUSSION TITLE
Multiple color attachments

### DISCUSSION THREAD

**Discussion by dragostis** [bevyengine/bevy#5794] [Q&A] [answered]
I'm trying to find a way to render all geometry into a side texture, `R32Uint`, where all pixels contain the entity ID of the corresponding entity that's drawn there. However, I want to do this without replacing the pbr shader/material.

I was looking at `pbr.wgsl` and there doesn't seem to be any obvious way to inherit the logic so I can simply add an extra uniform (the entity ID) and an extra color attachment at location 1.

Even though it may not be ideal, I was also looking at simply adding an extra `Material` to all entities and a second camera, but this did not work because both cameras were trying to render all both materials and I have not found a way to contain the materials per camera.

**Comment by superdump** [comment, marked as answer]
Take a look at the array texture example for how to reuse and extend the PBR shader functionality. You will basically have to copy and paste pbr.wgsl and add and use your uniform binding to it, make a custom material, specialise the pipeline to modify your layout. Feel free to come and ask on discord. I think we probably need a more complete / extensive example.

**Reply by dragostis** [reply]
Thanks for the answer. This could probably be made simpler by moving the PBR code from the fragment into an importable function.

Also, not sure how I can add multiple color attachments to a `Camera`.
